grails-app/services/org/pih/warehouse/inventory/CycleCountService.groovy

```diff
@@ -514,7 +514,7 @@ class CycleCountService {
         cycleCountItem.properties = command.properties
         cycleCountItem.countIndex = command.recount ? 1 : 0
         cycleCountItem.status = command.recount ? CycleCountItemStatus.INVESTIGATING : CycleCountItemStatus.COUNTING
-        cycleCountItem.dateCounted = new Date()
+        cycleCountItem.dateCounted = command.dateCounted ?: new Date()
```

Comment:
If command.dateCounted isn't provided but cycleCountItem.dateCounted already has a value, we probably don't want to override that with a new date.

Because the assigment is already happening at line 513:
```
cycleCountItem.properties = command.properties
```

I'd change this line to something like:
```
if (cycleCountItem.dateCounted = null) {
    cycleCountItem.dateCounted = new Date()
}
```

Reply:
Also, I see you added the dateCounted field to the create item call too but aren't using that on the backend.

In `createCycleCountItem(CycleCountItemCommand)` you'll want to add dateCounted to CycleCountItemCommand and then do:
```
CycleCountItem cycleCountItem = new CycleCountItem(
    ...
    dateCounted: command.dateCounted ?: new Date()
)
```